### PR TITLE
update share on password change

### PR DESCRIPTION
**Pull request by ArtificialOwl**
- remove password on previous share if enforce password protection is disabled (globally, not just on current circle)
- update password on previous shares on static password changes (if password protection is enabled)
- adding '--test-password' option in ./occ circles:manage:setting to confirm password protection of shares
- notice log on getShareByToken if password protected, including hashed password

### CODE REVIEW

File: lib/ShareByCircleProvider.php [hunk 1]
```diff
@@ -155,7 +124,7 @@ public function __construct(
 		$this->userManager = $userManager;
 		$this->rootFolder = $rootFolder;
 		$this->l10n = $l10n;
-		$this->logger = $logger;
+		$this->logger = OC::$server->get(LoggerInterface::class);
```
Comment (mejo-) [inline lib/ShareByCircleProvider.php:127]: Any reason to not use dependency injection here?

Comment (ArtificialOwl) [reply lib/ShareByCircleProvider.php:127]: legacy code on core :]

Comment (mejo-) [reply lib/ShareByCircleProvider.php:127]: Ok, but then you don't have to pass `$logger` into `__construct()`, do you?

**Review by mejo- — APPROVED** [review]
Looks good codewise appart from my comments.

File: lib/ShareByCircleProvider.php (hunk 1 above)
Comment (ArtificialOwl) [reply lib/ShareByCircleProvider.php:127]: i don't want to touch legacy code right now, might fix for nc26

**Pull request merged**